packages/fuels-core/src/code_gen/custom_types.rs

```diff
@@ -19,477 +19,639 @@ pub use utils::{
+                    ..Default::default()
+                },
+            ),
+            (
+                2,
+                TypeDeclaration {
+                    type_id: 2,
+                    type_field: String::from("bool"),
+                    ..Default::default()
+                },
+            ),
+        ]
+        .into_iter()
+        .collect::<HashMap<_, _>>();
+        let actual = expand_custom_enum(&p, &types)?.to_string();
+        let expected = TokenStream::from_str(
+            r#"
+            # [derive (Clone , Debug , Eq , PartialEq)] pub enum MatchaTea < > { LongIsland (u64) , MoscowMule (bool) } impl < > Parameterize for MatchaTea < > { fn param_type () -> ParamType { let mut param_types = vec ! [] ; param_types . push (< u64 > :: param_type ()) ; param_types . push (< bool > :: param_type ()) ; let variants = EnumVariants :: new (param_types) . unwrap_or_else (| _ | panic ! ("{} has no variants which isn't allowed!" , "MatchaTea")) ; ParamType :: Enum { variants , generics : vec ! [] } } } impl < > Tokenizable for MatchaTea < > { fn from_token (token : Token) -> Result < Self , SDKError > where Self : Sized , { let gen_err = | msg | { SDKError :: InvalidData (format ! ("Error while instantiating {} from token! {}" , "MatchaTea" , msg)) } ; match token { Token :: Enum (selector) => { let (discriminant , variant_token , _) = * selector ; match discriminant { 0u8 => Ok (Self :: LongIsland (< u64 > :: from_token (variant_token) ?)) , 1u8 => Ok (Self :: MoscowMule (< bool > :: from_token (variant_token) ?)) , _ => Err (gen_err (format ! ("Discriminant {} doesn't point to any of the enums variants." , discriminant))) , } } _ => Err (gen_err (format ! ("Given token ({}) is not of the type Token::Enum!" , token))) , } } fn into_token (self) -> Token { let (discriminant , token) = match self { Self :: LongIsland (inner) => (0u8 , inner . into_token ()) , Self :: MoscowMule (inner) => (1u8 , inner . into_token ()) } ; let variants = match Self :: param_type () { ParamType :: Enum { variants , .. } => variants , other => panic ! ("Calling {}::param_type() must return a ParamType::Enum but instead it returned: {}" , "MatchaTea" , other) } ; Token :: Enum (Box :: new ((discriminant , token , variants))) } } impl < > TryFrom < & [u8] > for MatchaTea < > { type Error = SDKError ; fn try_from (bytes : & [u8]) -> Result < Self , Self :: Error > { try_from_bytes (bytes) } } impl < > TryFrom < & Vec < u8 >> for MatchaTea < > { type Error = SDKError ; fn try_from (bytes : & Vec < u8 >) -> Result < Self , Self :: Error > { try_from_bytes (& bytes) } } impl < > TryFrom < Vec < u8 >> for MatchaTea < > { type Error = SDKError ; fn try_from (bytes : Vec < u8 >) -> Result < Self , Self :: Error > { try_from_bytes (& bytes) } }
+            "#,
+        )?
+        .to_string();
+
+        assert_eq!(actual, expected);
+        Ok(())
+    }
+
+    #[test]
+    fn top_lvl_enum_w_no_variants_cannot_be_constructed() -> anyhow::Result<()> {
```

Comment:
A lot has changed since these tests were written.

Due to the recursive nature of the type resolving, I'd say we're safe enough if we just cover that **a** enum cannot be expanded if it has no variants.

The same should hold for structs.

So maybe to ditch these 'nested enums/structs' tests and replace them with two tests. One for no-variant enums and another for no-field structs.